cetmix_tower_server/migrations/14.0.0.4.0/pre-migration.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from psycopg2.extensions import AsIs
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    """
+    Populate the `reference` field for tables that belong to
+    models inheriting from `cx.tower.reference.mixin`.
+    This will copy the value from the `name` field to the
+    `reference` field, but only for rows where `reference`
+    is empty.
+    """
+    _logger.info("Starting SQL migration for reference field population.")
+
+    # Step 1: Identify models that have a `reference` field
+    # and belong to `cx.tower` models
+    cr.execute(
+        """
+        SELECT m.model, f.model_id
+        FROM ir_model m
+        JOIN ir_model_fields f ON m.id = f.model_id
+        WHERE f.name = 'reference' AND m.model LIKE 'cx.tower%'
+    """
```

Comment:
strange way, you have to explicitly check that the mixin was inherited. e.g.:

```python
for model_name in self.env["ir.module.module"].search([("state", "=", "installed")]).mapped("name")
        model = request.env[model_name]
        if isinstance(model, self.env.registry['cx.tower.reference.mixin']):
            ....
```

Reply:
Thanks for review
I'll check it )